### ISSUE TITLE
Finance traces create a range slider by default but don't clear it on `deleteTraces`

### ISSUE THREAD

**Issue by alexcjohnson** [plotly/plotly.js#1473]
- Open http://localhost:3000/devtools/test_dashboard/#ohlc_first
- `Plotly.deleteTraces(gd,[0])`
- there's still data in the range slider:
![screen shot 2017-03-13 at 11 45 18 pm](https://cloud.githubusercontent.com/assets/2678795/23885060/27210340-0847-11e7-8dfa-b4f60f3fef8a.png)

ohlc_first.json says nothing about a range slider, it was just created by default. So there are two pieces to this:
1. the data is gone from the graph, it should disappear from the range slider too. If you delete any but the last trace, the range slider does update correctly, but not when you delete the last one.
2. should the range slider itself be removed when all data is gone from it? Perhaps, perhaps not, though I discovered this because there's a test in the workspace that expects it will disappear...

@etpinard the first part is a clear bug, what do you think about the second part though?

**Comment by etpinard** [comment]
> should the range slider itself be removed when all data is gone from it?

My gut feeling says _no_. 

```js
Plotly.newPlot('graph', [{
}], {
  xaxis: {rangeslider: {}}
})
```

currently outputs

![image](https://cloud.githubusercontent.com/assets/6675409/23902759/9430d4a0-0898-11e7-9eaf-a88d6c1a7e83.png)

which looks right, a range slider can exist without data attach to it. 

I believe I used the word [_orphan_](https://github.com/plotly/plotly.js/pull/289) to describe trace-less cartesian axes. That concept sounds extendable to range slider too.

**Comment by etpinard** [comment]
But @alexcjohnson in https://github.com/plotly/plotly.js/pull/289#issuecomment-190229241 regarding the orphan logic wrote:

> but given the convoluted logic that can arise I'm starting to think it's clearer for everyone if orphan subplots are simply not drawn.

so maybe my gut feeling is wrong.

**Comment by alexcjohnson** [comment]
No, I think your gut feeling is right - when you make a plot with no data but an explicit range slider, it should show up. I could imagine someone building an application that starts out looking like this and then gets data added to it, and they might think it looks better this way than with the range slider appearing with the first data, so they should have that option.

My concern was that `addTrace` and then `deleteTrace` does not get you back to where you started, as the range slider doesn't disappear. But that's starting to seem like it's too complicated for what is frankly a strange edge case. The only place I'm worried about it still is in the workspace, if someone makes a chart that automatically creates a range slider, then decides they don't want that trace type and deletes it, then doesn't know how to get rid of the range slider. But perhaps it's better to worry about that in the workspace, as it's really only an issue there, rather than add more complexity here.

**Comment by etpinard** [comment]
> My concern was that addTrace and then deleteTrace does not get you back to where you started, as the range slider doesn't disappear.

Correct. But that's only the case for `ohlc` and `candlestick` traces at the moment. So yeah I'd call that an edge case. Still, there might be a (clean and easy) way to make:

```js
Plotly.plot(gd, [])

Plotly.addTraces(gd, [{
  type: 'ohlc',
  /* */
]})

Plotly.deleteTraces(gd, [0])
```

remove the slider. I'll investigate.

**Comment by etpinard** [comment]
> the data is gone from the graph, it should disappear from the range slider too. If you delete any but the last trace, the range slider does update correctly, but not when you delete the last one.

fixed in #1472 

Renaming this issue to reflect item 2

> should the range slider itself be removed when all data is gone from it? Perhaps, perhaps not, though I discovered this because there's a test in the workspace that expects it will disappear...

**Comment by gvwilson** [comment]
Hi - this issue has been sitting for a while, so as part of our effort to tidy up our public repositories I'm going to close it. If it's still a concern, we'd be grateful if you could open a new issue (with a short reproducible example if appropriate) so that we can add it to our stack. Cheers - @gvwilson
